src/librustc/ty/layout.rs

```diff
@@ -833,7 +971,9 @@ pub enum Layout {
         nndiscr: u64,
         nonnull: Struct,
         // N.B. There is a 0 at the start, for LLVM GEP through a pointer.
-        discrfield: FieldPath
+        discrfield: FieldPath,
+        // Like discrfield, but in source order. For debuginfo.
+        discrfield_source: FieldPath
```

Comment:
I already discussed this with them via IRC when I originally wrote the solution.